index.rst

```diff
@@ -10,16 +10,200 @@
+The Prompt Processing system will be responsible for processing roughly a thousand visits per night, and distributing the results in near real time, for at least ten years of Rubin Observatory operations.
+As such, it must be highly robust to algorithmic, network, and infrastructure failures, ranging from momentary glitches to extended downtimes.
+`DMTN-219`_ introduced the initial design for the Prompt Processing framework; this document expands on the design to address expected failure modes and recovery strategies for each.
+
+.. _DMTN-219: https://dmtn-219.lsst.io/
+
+.. _pp-overview:
+
+Prompt Processing System Overview
+=================================
+
+The Prompt Processing system will receive images from both LATISS and LSSTCam, and run one of several pipelines on them, depending on the purpose of the observations.
+In the baseline case (running the full Alert Production pipeline on images for the Large Survey of Space and Time), the system will create multiple outputs: the transient alert stream, the alert production database (APDB), datasets in a central repository, and Sasquatch metrics for pipeline analytics and debugging.
+
+As of May 2023, the Prompt Processing system starts processing when it receives a ``next_visit`` event from the summit before observations are taken.
+The event is received by a fan-out service, which forwards a derived event for each detector to a pool of workers.
+Each fanned-out event represents the exposure(s) for a given visit with a given detector.
+Because the workers handle their assigned exposures independently and in parallel, this framework is efficient and flexible with respect to unexpected computational demands.
+However, it's impossible to make assumptions about the order of events between workers, and any given worker may be created or destroyed at any time.
+
+Each of the several hundred workers maintains an internal ("local") Butler repository for its processing; this keeps the central repository from becoming a performance bottleneck for pipeline execution.
+On receiving its visit-detector assignment, each worker downloads any calibrations, templates, etc. it will need but does not yet have from the central repo.
+The worker then waits for the raw exposures to arrive at USDF (or confirms they already arrived) and downloads and ingests them.
+When all exposures have arrived, the worker runs the pipeline, updating the APDB and sending alerts as appropriate; if not all exposures arrive, it attempts to process what it has.
```

Comment:
How does it detect when to give up?  Must be a timeout, right?

Reply:
In practice, probably. But is it a relevant detail in what is already a fairly verbose section?

Reply:
I think it's somewhat important to call it out, as pipeline-level timeouts may interact with worker-level timeouts.